### PR TITLE
Add task solutions

### PR DESCRIPTION
**Pull request by pankominek**

### CODE REVIEW

File: dom-basics-is-disabled.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const isDisabled = button => {
+    const disabled = button.hasAttribute('disabled');
```
Comment (IwoHerka) [inline dom-basics-is-disabled.js]: 1. Bad formatting. Properly align code so it's easier to read.
2. You can just simply do `return !disabled`.

File: dom-fetch-list-chapters.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import {Backend} from "./backend.js";
```
Comment (IwoHerka) [inline dom-fetch-list-chapters.js]: Missing empty spaces around the imported name:
```
import { Backend } from ...
```

File: dom-fetch-list-chapters.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const API = new Backend();
```
Comment (IwoHerka) [inline dom-fetch-list-chapters.js]: One empty line after imports.

File: dom-fetch-list-chapters.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    const button = document.querySelector('#list-chapters');
+    button.addEventListener("click", () => {
+        const list = document.querySelector('#chapters');
+        API.get("chapters/all.json").then(data => {
+            data.forEach(element => 
+                list.insertAdjacentHTML("beforeend", `<li class="item">${element.title}</li>`)
+            );
+        });
+    });
+
```
Comment (IwoHerka) [inline dom-fetch-list-chapters.js]: Remove this line.

File: dom-fetch-list-chapters.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import {Backend} from "./backend.js";
+const API = new Backend();
+API.setBaseUrl("https://api.learnjavascript.online/demo/");
+
+const initListChapters = () => {
+    const button = document.querySelector('#list-chapters');
+    button.addEventListener("click", () => {
```
Comment (IwoHerka) [inline dom-fetch-list-chapters.js]: One empty line before opening new lexical block.

File: dom-fetch-list-chapters.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+        const list = document.querySelector('#chapters');
+        API.get("chapters/all.json").then(data => {
```
Comment (IwoHerka) [inline dom-fetch-list-chapters.js]: Same here.

File: fetch-check-notifications.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const checkForNewNotifications = () => {
+    fetch('https://api.learnjavascript.online/demo/notifications/new.json')
```
Comment (IwoHerka) [inline fetch-check-notifications.js]: Bad alignment. Should be:

```
    fetch('https://api.learnjavascript.online/demo/notifications/new.json')
        .then(response => response.json())
        .then(data => displayNewNotifications(data.count));
```

File: project-iv-todo-item-web-component.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+class TodoItem extends HTMLElement {
+    constructor() {
+        super();
+    }
+
+     connectedCallback() {
+        const shadowRoot = this.attachShadow({mode: 'open'});
+        const template = document.querySelector('#todo-template');
+        shadowRoot.appendChild(template.content.cloneNode(true));
```
Comment (IwoHerka) [inline project-iv-todo-item-web-component.js]: Variable bindings first, then do everything else. This line should be lower.

**Comment by InternMakimo** [comment]
@IwoHerka, I corrected files according to your instructions.

**Comment by pankominek** [comment]
@IwoHerka, I corrected files according to your instructions.

**Comment by pankominek** [comment]
@IwoHerka, I corrected files according to your instructions.

File: dom-fetch-list-chapters.js (hunk 4 above)
Comment (pankominek) [reply dom-fetch-list-chapters.js]: I deleted this unnecessary line.

File: project-iv-todo-item-web-component.js (hunk 8 above)
Comment (pankominek) [reply project-iv-todo-item-web-component.js]: Ok, I changed the order of the lines.

File: fetch-check-notifications.js (hunk 7 above)
Comment (pankominek) [reply fetch-check-notifications.js]: Corrected.

File: dom-basics-is-disabled.js (hunk 1 above)
Comment (pankominek) [reply dom-basics-is-disabled.js]: I'm not sure I solved it right.

File: dom-fetch-list-chapters.js (hunk 2 above)
Comment (pankominek) [reply dom-fetch-list-chapters.js]: Corrected.

File: dom-advenced-add-item-to-shopping-list.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+const addItemToShoppingList = item => {
+    if(item) {
```
[the remark this answers is not in the document]
Comment (pankominek) [reply dom-advenced-add-item-to-shopping-list.js]: Corrected.

File: async-await-async-await-usage.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
+class Backend {
+    constructor() {
+        this.baseUrl = '';
+    }
```
[the remark this answers is not in the document]
Comment (pankominek) [reply async-await-async-await-usage.js]: Corrected.

File: dom-fetch-list-chapters.js (hunk 2 above)
Comment (pankominek) [reply dom-fetch-list-chapters.js]: Corrected.

File: project-iv-todo-item-web-component.js (hunk 8 above)
Comment (InternMakimo) [reply project-iv-todo-item-web-component.js]: Corrected.

File: dom-advenced-show-GDPR-consent.js [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const updateFormEUConsent = user => {
+    if (user.is_eu) {
```
[the remark this answers is not in the document]
Comment (pankominek) [reply dom-advenced-show-GDPR-consent.js]: I added camel-case variable. Sample usage:

updateFormEUConsent({
    id: 1,
    is_eu: true
});

updateFormEUConsent({
    id: 2,
    is_eu: false
});

File: functions-II-lexical-scope.js [hunk 12]
```diff
@@ -0,0 +1,16 @@
+class User {
+  constructor(age) {
```
Comment (IwoHerka) [inline functions-II-lexical-scope.js:2]: Everywhere you use 4 spaces, but here are 2s. Why?

File: import-export-import-helper.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+import {Helper} from "./helper.js";
```
Comment (IwoHerka) [inline import-export-import-helper.js:1]: Spaces.

File: project-II-API-wrapper.js [hunk 14]
```diff
@@ -0,0 +1,49 @@
+class Backend {
+  constructor() {
```
Comment (IwoHerka) [inline project-II-API-wrapper.js:2]: Here too you are using 2 spaces only. Should be 4.

File: import-export-import-helper.js (hunk 13 above)
Comment (pankominek) [reply import-export-import-helper.js]: Corrected.

File: functions-II-lexical-scope.js (hunk 12 above)
Comment (pankominek) [reply functions-II-lexical-scope.js]: Corrected. I always use the tab button to get space.

File: project-II-API-wrapper.js (hunk 14 above)
Comment (pankominek) [reply project-II-API-wrapper.js]: Corrected.

**Pull request merged**